### ISSUE TITLE
Upgrading to 2.2.0 breaks being able to map to EF objects

### ISSUE THREAD

**Issue by bmccord** [AutoMapper/AutoMapper#253]
I am using EF with the Code Generation Item called EF 5.x DbContext Generator.  It basically generates a class that looks like this:

```
public partial class Department
{
    public Department()
    {
        this.Employments = new HashSet<Employment>();
    }

    public short DepartmentKey { get; set; }
    public string DepartmentName { get; set; }
    public int CreatedBy { get; set; }
    public System.DateTime CreatedDateTime { get; set; }
    public int LastModifiedBy { get; set; }
    public System.DateTime LastModifiedDateTime { get; set; }

    public virtual ICollection<Employment> Employments { get; set; }
    public virtual Person Person { get; set; }
    public virtual Person Person1 { get; set; }
}
```

I am mapping from this to a class that looks like this:

```
public class Department
{
    public int DepartmentKey { get; set; }
    public string DepartmentName { get; set; }
    public bool _destroy { get; set; }
    public bool hasChanges { get; set; }
}
```

Mapping from the EF class to the smaller class works just fine, but when trying to save data and go in the reverse direction it doesn't work.  It tells me the mapping doesn't exist, but it does.

When I save, I am doing this:

```
                var departmentToSave = database.Departments.FirstOrNew<EF.Department>(o => o.DepartmentKey == department.DepartmentKey);

                if (department.DepartmentKey == 0)
                {
                    departmentToSave.CreatedBy = 0;
                    departmentToSave.CreatedDateTime = DateTime.Now;
                }

                departmentToSave.LastModifiedBy = 0;
                departmentToSave.LastModifiedDateTime = DateTime.Now;
                Mapper.Map<Department, EF.Department>(department, departmentToSave);
                database.Departments.AddOrUpdate(departmentToSave);
```

Basically, EF doesn't return the exact object, but a proxy.  However, this mapping worked just fine in 2.1.266.  Upgrading to 2.2.0 causes this to fail.

I have backed up to 2.1.266 and it works again.

Is this by design, and if so, what should I do differently?
